library/common/extensions/filter/platform_filter.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "envoy/http/filter.h"
+
+#include "extensions/filters/http/common/pass_through_filter.h"
+
+#include "library/common/extensions/filter/c_types.h"
+
+namespace Envoy {
+namespace Http {
+/**
+ * Harness to bridge Envoy filter invocations up to the platform layer.
+ */
+class PlatformExtensionFilter final : public PassThroughFilter {
+public:
+  PlatformExtensionFilter(envoy_http_filter platform_filter) : platform_filter_(platform_filter) {}
+
+  // StreamDecoderFilter
+  FilterHeadersStatus decodeHeaders(RequestHeaderMap& headers, bool end_stream) override;
```

Comment:
For now we don't need any of the other overrides except this one. We want to attempt to  harness a call to decode headers to modify the request header, right? The `PassThroughFilter` takes care of the rest.

Reply:
We don't need them for a prototype, but I left them here for reference (and if we go this route, we'll definitely have them).

Reply:
Yeah, but given that the pass through filter implements, I was wondering if we should delete to make this PR easier to read TIOLI

Reply:
Either way.